packages/atlas/src/components/FilterButton/FilterButton.tsx

```diff
@@ -13,85 +15,157 @@ export type FilterButtonOption = {
-    const selected = options.map((option, idx) => {
-      if (num === idx) {
-        return { ...option, selected: !option.selected }
-      }
-      return option
-    })
-    onChange(selected)
-  }
+    const counter = options.filter((option) => option.applied)?.length
+
+    const handleApply = () => {
+      onChange(options.map((option) => ({ ...option, applied: option.selected })))
+      triggerRef.current?.click()
+    }
+
+    const handleCheckboxSelection = (num: number) => {
+      const selected = options.map((option, idx) => {
+        if (num === idx) {
+          return { ...option, selected: !option.selected }
+        }
+        return option
+      })
+      onChange(selected)
+    }
 
-  const handleRadioButtonClick = (e: ChangeEvent<Omit<HTMLInputElement, 'value'> & { value: string | boolean }>) => {
-    const optionIdx = options.findIndex((option) => option.value === e.currentTarget.value)
-    const selected = options.map((option, idx) => ({ ...option, selected: optionIdx === idx }))
-    onChange(selected)
+    const handleRadioButtonClick = (e: ChangeEvent<Omit<HTMLInputElement, 'value'> & { value: string | boolean }>) => {
+      const optionIdx = options.findIndex((option) => option.value === e.currentTarget.value)
+      const selected = options.map((option, idx) => ({ ...option, selected: optionIdx === idx }))
+      onChange(selected)
+    }
+
+    const handleClear = () => {
+      onChange(options.map((option) => ({ ...option, selected: false, applied: false })))
+      triggerRef.current?.click()
+    }
+
+    return (
+      <DialogPopover
+        className={className}
+        flipEnabled
+        appendTo={document.body}
+        trigger={
+          <StyledButton
+            ref={triggerRef}
+            icon={counter ? <Counter>{counter}</Counter> : icon}
+            iconPlacement="right"
+            variant="secondary"
+          >
+            {label}
+          </StyledButton>
+        }
+        primaryButton={{ text: 'Apply', onClick: handleApply }}
+        secondaryButton={{
+          text: 'Clear',
+          onClick: handleClear,
+        }}
+      >
+        {type === 'checkbox' && (
+          <CheckboxGroup
+            name={name}
+            options={options.map((option) => ({ ...option, value: option.selected }))}
+            checkedIds={options.map((option, index) => (option.selected ? index : -1)).filter((index) => index !== -1)}
+            onChange={handleCheckboxSelection}
+          />
+        )}
+        {type === 'radio' && (
+          <RadioButtonGroup
+            name={name}
+            options={options}
+            value={options.find((option) => option.selected)?.value}
+            onChange={handleRadioButtonClick}
+          />
+        )}
+      </DialogPopover>
+    )
   }
 
-  const handleClear = () => {
-    onChange(options.map((option) => ({ ...option, selected: false, applied: false })))
-    triggerRef.current?.click()
+  if (props.type === 'range') {
```

Comment:
I think it's nice to have: 
![image](https://user-images.githubusercontent.com/51168865/233059341-8fd6be11-31b2-4089-9243-de7f03f60b6e.png)